components/freertos/include/freertos/ringbuf.h

```diff
@@ -297,6 +297,18 @@ BaseType_t xRingbufferAddToQueueSetRead(RingbufHandle_t ringbuf, QueueSetHandle_
  */
 BaseType_t xRingbufferAddToQueueSetWrite(RingbufHandle_t ringbuf, QueueSetHandle_t xQueueSet);
 
+/**
+ * @brief Check if semaphore belongs to ringbuffer
+ *
+ * This API checks if Semaphore returned from xQueueSelectFromSet belongs to this ring buffer
+ * and if it is for read locking.
+ *
+ * @param ringbuf Ring buffer which should be checked
+ * @param sem Semaphore returned from xQueueSelectFromSet
+ *
+ * @return pdTRUE when semaphore belongs to ringbuffer, pdFALSE otherwise.
+ */
+BaseType_t xRingbufferCanRead(RingbufHandle_t ringbuf, QueueHandle_t sem);
```

Comment:
Suggest changing from ``QueueHandle_t sem`` to ``QueueSetMemberHandle_t member`` to conform with FreeRTOS queueset API